Frameworks/CoreMotion/CMMotionManager.mm

```diff
@@ -249,56 +321,140 @@ - (void)stopGyroUpdates {
 
  // getter function for on-demand reading from sensor
  -(CMGyroData*)gyroData {
+     
     WDSGyrometerReading* reading = [self.gyrometer getCurrentReading];
 
     // For gyrometer values iOS uses unit as radians per sec. while WinRT uses degrees per sec. 
-    _gyroData = [[CMGyroData alloc] initWithValues:reading.angularVelocityX * c_degreeToRadians 
-                                                 y:reading.angularVelocityY * c_degreeToRadians
-                                                 z:reading.angularVelocityZ * c_degreeToRadians];
+    _gyroData = [[CMGyroData alloc] initWithValues:reading.angularVelocityX * c_degreeToRadian 
+                                                 y:reading.angularVelocityY * c_degreeToRadian
+                                                 z:reading.angularVelocityZ * c_degreeToRadian
+                                              time:[self toSecondsSinceBoot:reading.timestamp]];
     return _gyroData;
  }
 
 
  -(void)setGyroUpdateInterval:(NSTimeInterval)updateInterval {
```

Comment:
The logic in this function could potentially be pathologically broken.  Ex: if mimimumReportInterval is 1010 ms, and updateInterval is 1sec, we will set _gyroUpdateInterval to 1sec and lateron try to update reportInterval to 1000ms, which is lower than 1010 ms.  In general, i think we should maintain the state in higher fidelity (in msec) and do the capping in the accessors.  Granted, this is not new code with this change.  Can we track that issue outside this PR?  Same issue in other methods that do similar math.